src/ui/settings/components/localeSetting.tsx

```diff
@@ -0,0 +1,86 @@
+import { StyleSheet, Text, View } from 'react-native'
+import { Colors, Spacing, Typography } from '../../../styles'
+import React from 'react'
+import SettingsItem from './settingsItem'
+import I18n from '../../../locales/i18n'
+import strings from '../../../locales/strings'
+
+const styles = StyleSheet.create({
+  languageOptions: {
+    flexDirection: 'row',
+    marginTop: Spacing.SM,
+  },
+  languageOption: {
+    justifyContent: 'center',
+    backgroundColor: Colors.lightGreyLight,
+    borderRadius: 4,
+    paddingHorizontal: Spacing.MD,
+    paddingVertical: Spacing.XS,
+    marginRight: Spacing.MD,
+  },
+  languageOptionText: {
+    ...Typography.baseFontStyles,
+    fontSize: Typography.textXS,
+    color: Colors.blackMain,
+  },
+})
+
+interface ComponentProps {
```

Comment:
You can also define an interface, i.e. `CommonProps` and extend it twice, would look like this:

```
interface CommonProps {
  currentLocale: string
  setLocale: (key: string) => void
}

interface Component props extends CommonProps {
  locales: string[]
}

interface CardProps extends CommonProps {
  locale: string
}
```

Reply:
Is it fine having an extra interface when they're not likely to be used anywhere else?

Reply:
> Is it fine having an extra interface when they're not likely to be used anywhere else?

You could ask the same question about the `LocaleCard` itself, which is only used once (and not expected to be used outside of here) and can be clearly inlined in `LocaleSetting`

IMHO it's all a matter of keeping things easy to handle/wrangle (so each "unit", of whatever it is we are considering, should be of a manageable size), without introducing too much "organizational complexity" or "boilerplate" or otherwise superfluous code. And there's quite a bit of subjectivity involved there :smile:

My subjective take on it:
- Avoid any "uncommon" types unless required. So, a clear inheritance hierarchy would be preferred over something using `Omit<>`
- Avoid unhelpful encapsulation of code that is too small to be an issue or to be reused, etc. So inline `LocaleCard` (and thus avoid extra types and also avoid burying logic such as `const isCurrentLanguage = locale === props.currentLocale` so deep inside the view layer. Stuff like that should remain at the higher levels, and the view layer should only be getting a "this one is highlighted" kind of input (i.e. it should not be deciding what to highlight, otherwise it needs to "understand" what these things are)